src/game_display.cpp

```diff
@@ -200,7 +200,7 @@ void game_display::scroll_to_leader(int side, SCROLL_TYPE scroll_type,bool force
 {
 	unit_map::const_iterator leader = dc_->units().find_leader(side);
 
-	if(leader.valid()) {
+	if(leader.valid() && leader->is_visible_to_team(dc_->teams()[viewing_team()], *dc_, false)) {
```

Comment:
I'd prefer `dc_->get_team(viewing_side())` instead.

Reply:
Changed. What's the difference? Bounds checking on the vector element access?

Reply:
Exactly.